pkg/sharky/store.go

```diff
@@ -48,22 +49,25 @@ func New(basedir string, shardCnt int, limit uint32, datasize int) (*Store, erro
 	pool := &sync.Pool{New: func() interface{} {
 		return make(chan entry)
 	}}
-	sh := &Store{
+	store := &Store{
 		datasize: datasize,
 		pool:     pool,
 		writes:   make(chan write),
 		shards:   make([]*shard, shardCnt),
 		wg:       &sync.WaitGroup{},
 		quit:     make(chan struct{}),
+		metrics:  newMetrics(),
 	}
-	for i := range sh.shards {
-		s, err := sh.create(uint8(i), limit, datasize, basedir)
+	for i := range store.shards {
+		s, err := store.create(uint8(i), limit, datasize, basedir)
 		if err != nil {
 			return nil, err
 		}
-		sh.shards[i] = s
+		store.shards[i] = s
 	}
-	return sh, nil
+	store.metrics.ShardSize.Set(float64(len(store.shards)))
```

Comment:
just to mention, this is not `shard size` but `number of shard`. shard size would be the number of the items in the shard, which would be the `filesize/datasize` and should be updated when the shard grows

Reply:
Done.